src/python/pants/backend/python/target_types.py

```diff
@@ -1333,6 +1333,7 @@ class PythonDistribution(Target):
     alias = "python_distribution"
     core_fields = (
         *COMMON_TARGET_FIELDS,
+        InterpreterConstraintsField,
```

Comment:
It is mind-boggling that this is literally the only change needed to allow users to specify interpreter_constraints directly on a python_distribution and it all just works!

Reply:
This tells me that we did something very right with the target API. Literally just this, and all the plumbing magically happens.

Reply:
And "we" means @Eric-Arellano

Reply:
+1 much much prior design thought and review feedback from @stuhood and friends :) 

Yay Field-Driven API!